### ISSUE TITLE
#8399 T4b build 210bb03f2f Filter Field Disappears Recurring & Intermittent 

### ISSUE THREAD

**Issue by chaseholden** [transmission/transmission#4040]
### What is the issue?

#8399 build 210bb03f2f Transmission 4 beta 

- The Filter Field Disappears completely such that only its bar of space appears empty, the field is absent. Screencap video included.

- This has been a Recurring issue that has persisted for many months for me but I have not reported it because I presumed it was a known issue. 

- The issue is also Intermittent and eventually re-appears after I play with toggling view of the status bar and toggling view of filter bar, and even hiding the entire T4 window and showing it again. Sometimes it seems like I have to show/hide Groups but I can't be certain. 

- The point is I can't figure out what the pattern to reproduce it other than that the Filter input Field is often just missing and I toggle different views to get it back eventually. 

- Note that in order to reduce CPU use and improve bandwidth performance, I very frequently Hide Status Bar, Collapse all Groups, and Filter Torrents on an unused character like "=" in order to hide everything possible, using keyboard shortcuts ⌘B, ⌘F - (no I don't think use keyboard shortcuts are related.)

https://user-images.githubusercontent.com/19694741/198183996-0a466010-cea6-482a-b6d7-fef57c17e02d.mp4

### Which application of Transmission?

macOS app

### Which version of Transmission?

#8399 T4b build 210bb03f2f

**Comment by Coeur** [comment]
@chaseholden @Tockman if you can build from source, and if you somehow are able to reproduce the issue semi-consistently, can you try to add this declaration in FilterButton.mm?
```
- (NSSize)intrinsicContentSize
{
    return super.intrinsicContentSize;
}
```
Normally it shouldn't do anything, but maybe it helps?

**Comment by Coeur** [comment]
As I can't reproduce the issue myself, I can't certify the origin of the issue or how to fix it.

Another possibility: maybe it's caused by the vertical line which has:
- an horizontal content hugging priority of 750 (the other elements of this bar have it at 250)
- a constraint for a width of 1 when the width is normally non-editable at 5

Those strangenesses may be causing unexpected behaviour.
![Capture d’écran 2022-11-02 à 21 38 22](https://user-images.githubusercontent.com/839992/199504680-94e8da26-6b1e-4409-afb2-7cdbcea7a99a.png)

**Comment by Tockman** [comment]
![Снимок экрана 2022-12-05 в 18 11 33](https://user-images.githubusercontent.com/74548419/205673263-43759691-82ca-463d-9f32-4545c17311cb.png)
T4 b2 (bceb368f1b): problem still persists.

**Comment by Coeur** [comment]
Ah, then it's possibly a duplicate of #3400 I would say. :)

**Comment by Tockman** [comment]
A very interesting modification of this error I observe now with beta 3. At some point Filter buttons get inactive and stop to respond on mouse clicks. But: if you maximize Transmission window and then minimize it back, they become active and working again.

**Comment by chaseholden** [comment]
This continues to be a bug in 4.0.2

**Comment by jeremyelgin** [comment]
I've just switched back to this client, using 4.0.3 and noticed this.
I've been restarting the app (or rebooting) because I haven't yet worked out any other fix, but it reoccurs.

**Comment by Coeur** [comment]
Marking as duplicate of #3400, but keeping it open as it has a nice video of the issue, showing a slightly different situation than the previous issue.

**Comment by chaseholden** [comment]
I believe this has been solved for me as of version 4.0.5. I have not experienced it for several months. I am going to close this but if anyone experiences this in 4.0.5 or later, please reopen it.
